File: bukkit/modules/autorespawn/src/main/java/dialight/autorespawn/AutoRespawnModule.java
package dialight.autorespawn;

import dialight.modulelib.Module;

public class AutoRespawnModule extends Module {

    public static final String ID = "autorespawn";

    private final AutoRespawn proj;

    public AutoRespawnModule(AutoRespawn proj) {
        super();
        this.proj = proj;
    }

    @Override
    public String getId() {
        return ID;
    }

}

File: modules/autorespawn/src/main/kotlin/dialight/autorespawn/AutoRespawnModule.kt
package dialight.autorespawn

import dialight.modulelib.AbstractModule
import org.spongepowered.api.Sponge

class AutoRespawnModule(val plugin: AutoRespawnPlugin) : AbstractModule(AutoRespawnModule.ID, "AutoRespawn") {

    companion object {
        val ID = "autorespawn"
    }

    override fun onEnable(): Boolean {
        Sponge.getEventManager().registerListeners(plugin, plugin.listener)
        return true
    }

    override fun onDisable(): Boolean {
        Sponge.getEventManager().unregisterListeners(plugin.listener)
        return true
    }

}